src/gc.c

```diff
@@ -3669,29 +3669,34 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
     last_live_bytes = live_bytes;
     live_bytes += -gc_num.freed + actual_allocd;
 
+    // XXX: we've observed that the `live_bytes` was negative in a few cases
+    // which is not expected. We should investigate this further, but let's just
+    // cap it to 0 for now.
+    int64_t live_bytes_for_interval_computation = live_bytes < 0 ? 0 : live_bytes;
```

Comment:
Do we also want to cap it in the other direction? It should not exceed maximum memory, right?

Reply:
If I were to cap anything in the other direction (i.e. put an upper bound on anything), that would be the `gc_num.interval`, probably.

With this change, I believe it's already upper-bounded by `max_total_memory`, but we can possibly make that bound a bit tighter (i.e. `max_total_memory / 2` or so).

Reply:
If this is just a patch for v.1.10.2+RAI and not intended for upstream, could we limit gc_num.interval to `max_total_memory / log2(max_total_memory)` ?

Reply:
What's the motivation for dividing by log?

Reply:
To be more precise: why the choice of log here? And not some other function?

Reply:
The OOMGuardian heuristic triggers gc when the heap size increases by 15% over the high-water mark in a rolling window (10 minutes). This avoids having the heap be more than 15% larger than it needs to be. It also means that if the reachable heap size is increasing monotonically, we don't do more than a logarithmic number of collections. In the scenario where we don't know the actual heap size because of the accounting problem, I'm going for a logarithmic number of collections in physical memory size. The `max_total_memory / log2(max_total_memory)` calculation would give an interval of 0.5GiB for 16GiB ram (32 collections until all of physical memory allocated), and 6.7GiB for 256GiB ram (38 collections).

Reply:
Implemented in latest commit.